crates/hyperdrive-math/src/lp.rs

```diff
@@ -353,6 +412,164 @@ impl State {
             ))
             .unwrap()
     }
+
+    /// Calculates the number of share reserves that are not reserved by open
+    /// positions.
+    fn calculate_idle_share_reserves(&self) -> FixedPoint {
+        let long_exposure = self.long_exposure().div_up(self.vault_share_price());
+        let idle_shares = {
+            if self.share_reserves() > long_exposure + self.minimum_share_reserves() {
+                return self.share_reserves() - long_exposure - self.minimum_share_reserves();
+            }
+            fixed!(0)
+        };
+
+        idle_shares
+    }
+
+    /// Note that the state is the present state of the pool and original values
+    /// passed in as parameters.  Present sate variables are not expressly
+    /// paased in because so that downstream function like kUp() can still be
+    /// used.
+    ///
+    /// Given a signed bond amount, this function calculates the negation
+    /// of the derivative of `calculateSharesOutGivenBondsIn` when the
+    /// bond amount is positive or the derivative of
+    /// `calculateSharesInGivenBondsOut` when the bond amount is negative.
+    /// In both cases, the calculation is given by:
+    ///
+    ///  derivative = (1 - zeta / z) * (
```

Comment:
can you change `x` to `delta_y` (or `\Delta y` if in latex)? We've been somewhat inconsistent about this elsewhere, I'm trying to move us in the direction of consistently using `x` and `\Delta x` for "base".

Reply:
assuming I have that right... seeing `y(x)` confuses me further.... Maybe the request is better stated as "can you make sure we use `z` for shares; `y` for bonds; and `x` for base)

Reply:
same thing here, I wanna keep it exactly the same as the solidity to avoid drift.  love that you're looking at these though (and understanding haha).  can we keep these comment changes outta this PR and then we can sit down and update in both spots?  🙏

Reply:
yup sounds good. We already have issue for it:
https://github.com/delvtech/hyperdrive/issues/867
https://github.com/delvtech/hyperdrive/issues/965

can you link 965 in a TODO comment here?